README.md

```diff
@@ -554,42 +467,120 @@ NOTE: Schema for `response`
 }
 ```
 
+### Component
+
+The named `component` import provides controls of the modal component.
+
+```js
+import { component } from 'forgerock-login-widget';
+
+// Initiate the component API
+const componentEvents = component();
+
+// Know when the model has been mounted or closed to run your own logic
+// The property `reason` will be either "auto", "external", or "user" (see below)
+componentEvents.subscribe((event) => {
+  /* Run anything you want */
+});
+
+// Open the modal
+componentEvents.open();
+
+// Close the modal
+componentEvents.close();
+```
+
+Schema for component `event`:
+
+```js
+{
+  error: null, // null or object with `code`, `message` and `step` that failed
+  mounted: false, // boolean
+  open: null, // boolean or null, depending on the Widget type: e.g. "modal" or "inline"
+  reason: null, // string to describe the reason for the event
+  type: null, // 'modal' or 'inline'
+}
+```
+
+The `reason` value is used for communicating why the modal has closed. The below are the potential values:
+
+1. `"user"`: user closed the dialog via UI
+2. `"auto"`: the modal was closed because user successfully authenticated
+3. `"external"`: the application itself called the `modal.close` function
+
 ### User
 
-The `user` object:
+As with almost all of our APIs, there's a choice between using Observables or Promises. We recommend Observables, but the choice is up to you. We will cover the Promise alternative in a different section.
```

Comment:
By a "different section" I'm guessing it's a page that isn't in this PR, and that will be written up closer to release? I'm just wondering if we should have an example of using Promises and a brief description you might want to, e.g.:

_You may wish to interact with the API using Promises instead of Observables if this is what you're more comfortable with, and only need the "final state" of an API call. For example calling user.tokens() using Promises returns you the "final state" once the underlying requests have returned:_
```
      const userEvents: any = user.tokens();
      const userResult = await userEvents.get();
      if (userResult.completed && userResult.successful) {
        // do something based on the fact that the tokens call was successful
      }
```

I would worry that it feels like we are pushing observables and not providing details of the alternative, for developers / beta testers in a hurry. Thoughts?

Reply:
Add a more thoroughly explained section for this: https://forgerock-web-login-framework.vercel.app/docs/widget/full-api